src/app/features/landing/components/landing-hero/landing-hero.component.scss

```diff
@@ -0,0 +1,29 @@
+@import 'primeflex/primeflex.scss';
```

Comment:
Why do we have to import this here? Can we not do it in the main style.scss? As this is going to be common. Where does it get added by default(maybe in angular.json too)?

Reply:
### Background
We have imported prime flex to the `main.scss`. That allows us to use the utility classes to any HTML page.
I didn't wanted to mess up HTMLs with the utility classes so the idea was to use them in scss file.

### Observation
What I observed is the classes does not work until we import it to the particular style file. It's just like we import `mixins` in scss to every file before using, even if that is imported in the `main.scss` file.

Reply:
I had only used this once that too in one component.
And this is how it was used in `src/style.scss`

![image](https://github.com/OsmosysSoftware/angular-foundation/assets/14059393/a7ffc060-f3d4-4715-8dd9-2b244acd2152)

No other import was added.

Reply:
@xixas  Were you then able to use the utility classes in the scss file with out importing them?

By utility classes I mean the classes they provide and those being used with  `styleclass` mixin
```
@include styleclass('py-4 flex align-items-center justify-content-between relative lg:static mb-3');
```
To add context, the purpose we are solving here is using the primeflex classes in our style.scss files.

Reply:
I was able to use it directly. Not much used in my usecase, these were the classes I used.

![image](https://github.com/OsmosysSoftware/angular-foundation/assets/14059393/99c342b7-958e-46a2-b3be-dfcebd6a3f55)

Reply:
Yes, you are right. As I said, classes can be used in HTML by importing once.
But while using the classes in SCSS we will have to import them to each file to use the mixins.

Using the classes in HTML is not preferable since it will make things look messy. We can choose to have cleaner code by taking things to scss.

Reply:
@ashutosh-bisoyi  Is there any degradation in performance?
Becasue we are using complete @import 'primeflex/primeflex.scss' in all scss files

Checked with chatgpt and this was a suggested organisation.
https://chat.openai.com/share/90ea192f-f25a-4560-a221-7cf9a3cce627

```scss
// styles.scss (main entry point)
@import 'primeflex/primeflex.scss';
@import 'global-styles.scss';
@import 'component-styles.scss';

// global-styles.scss
// Your global styles, variables, and mixins

// component-styles.scss
// Import styles for specific components or modules
@import 'button.scss';
@import 'grid.scss';
// ...

// button.scss
// Styles for buttons using PrimeFlex classes
.button {
  @include p-button;
  // Additional styles or customizations
}

// grid.scss
// Styles for grid layout using PrimeFlex classes
.container {
  @include p-container;
  // Additional styles or customizations
}
```

Reply:
About performance, I am not really sure, but hope it should not be a problem. This is the way they suggest to use their classes in scss.
References:

1. [Official doc](https://primeflex.org/installation#reuseclasses)
2.  [discussion](https://github.com/primefaces/primeflex/issues/68#issuecomment-945282634)